internal/services/tesla_fleet_api_service.go

```diff
@@ -0,0 +1,141 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/rs/zerolog"
+	"golang.org/x/oauth2"
+)
+
+//go:generate mockgen -source tesla_fleet_api_service.go -destination mocks/tesla_fleet_api_service_mock.go
+type TeslaFleetAPIService interface {
+	CompleteTeslaAuthCodeExchange(authCode, redirectURI, region string) (*TeslaAuthCodeResponse, error)
+	GetVehicles(token, region string) ([]TeslaVehicle, error)
```

Comment:
Let's add a context to each one of these calls.

Reply:
We have a context for timeout within both functions as well.

Reply:
That's also good, but I think ideally the incoming request being killed should also kill the child requests going out.